### PR TITLE
Support ignoring routes or actions

### PR DESCRIPTION
**Pull request by distributedlife**
Added support for a .traceroute, .traceroute.yml or .traceroute.yaml file. If it's found in the root directory then it'll read the file in and add the ignore regexes to the array of routes/actions to ignore.

I've added some doco to the readme. Tests are included.

### CODE REVIEW

**Comment by lowjoel** [comment]
:+1:

**Comment by misteral** [comment]
:+1:

**Comment by MehdiK** [comment]
Is there any chance you can merge this PR and release it please? We have been pointing to the GitHub url from our project and just want to point to the gem. Thanks.

File: lib/traceroute.rb [hunk 1]
```diff
@@ -25,7 +26,7 @@ def defined_action_methods
       controller.action_methods.reject {|a| (a =~ /\A(_conditional)?_callback_/) || (a == '_layout_from_proc')}.map do |action|
         "#{controller.controller_path}##{action}"
       end
-    end.flatten.reject {|r| r.start_with? 'rails/'}
+    end.flatten.reject {|r| @ingored_unreachable_actions.any? { |m| r.match(m) } }
```
Comment (torrocus) [inline lib/traceroute.rb]: Here it is probably a typo `ignored` instead `ingored`

File: lib/traceroute.rb [hunk 2]
```diff
@@ -35,10 +36,51 @@ def routed_actions
       else
         "#{r.requirements[:controller]}##{r.requirements[:action]}"
       end
-    end.reject {|r| r.start_with? 'rails/'}
+    end.flatten.reject {|r| @ingored_unused_routes.any? { |m| r.match(m) } }
```
Comment (torrocus) [inline lib/traceroute.rb]: Typo as above

File: README.rdoc [hunk 3]
```diff
@@ -68,6 +68,20 @@ Running the Rake task will print something like this for you:
...
 OMG super helpful, isn't it?
 
+== How do I tell it to ignore routes?
+
+Some gems out there that inject routes or actions into our app for testing. Jasmine-Rails does this. They can give you false negatives when running this in development mode. It can be useful to ignore said routes or actions.
+
+Create a .traceroute.yaml (or .traceroute.yml or .traceroute) file in your root directory.
+
+  # .traceroute.yaml
+  ignore_unreachable_actions:
```
Comment (torrocus) [inline README.rdoc]: Inconsistency. Here is `ignore_unreachable_actions` and in `lib/traceroute.rb` is `@ingored_unreachable_actions`

File: README.rdoc [hunk 4]
```diff
@@ -68,6 +68,20 @@ Running the Rake task will print something like this for you:
...
+  - ^jasmine_rails\/
+  ignore_unused_routes:
```
Comment (torrocus) [inline README.rdoc]: As above

File: lib/traceroute.rb [hunk 5]
```diff
@@ -35,10 +36,51 @@ def routed_actions
...
+  end
+
+  def ignore_config
+    filenames.each do |filename|
+      return YAML.load_file(filename)
+    end
+  end
+
+  def load_ignored_regex!
+    @ingored_unreachable_actions = [/^rails\//]
```
Comment (torrocus) [inline lib/traceroute.rb]: Typo

File: lib/traceroute.rb [hunk 6]
```diff
@@ -35,10 +36,51 @@ def routed_actions
...
+    @ingored_unused_routes = [/^rails\//]
```
Comment (torrocus) [inline lib/traceroute.rb]: Typo

File: lib/traceroute.rb [hunk 7]
```diff
@@ -35,10 +36,51 @@ def routed_actions
...
+
+    return unless at_least_one_file_exists?
+
+    if ignore_config.has_key? 'ignore_unreachable_actions'
+      unless ignore_config['ignore_unreachable_actions'].nil?
+        ignore_config['ignore_unreachable_actions'].each do |ignored_action|
+          @ingored_unreachable_actions << Regexp.new(ignored_action)
```
Comment (torrocus) [inline lib/traceroute.rb]: Typo

File: lib/traceroute.rb [hunk 8]
```diff
@@ -35,10 +36,51 @@ def routed_actions
...
+        end
+      end
+    end
+
+    if ignore_config.has_key? 'ignore_unused_routes'
+      unless ignore_config['ignore_unused_routes'].nil?
+        ignore_config['ignore_unused_routes'].each do |ignored_action|
+          @ingored_unused_routes << Regexp.new(ignored_action)
```
Comment (torrocus) [inline lib/traceroute.rb]: Typo

**Comment by piotr-galas** [comment]
What about this pr?  I can help fix it if somebody will merge it.

**Comment by StephanMeijer** [comment]
Please. Merge.

**Comment by amatsuda** [comment]
Just published 0.6.0 gem with this feature.
But TBH I'm doubting if you guys still need this feature.

Another big feature intruduced in 0.6.0, Rails Engines support, would totally cover your Jasmine or other mounted Engines use case, so I guess most of you who showed up in this thread might not in fact need to deal with the YAML file thing.

Anyway, sorry for letting you wait so long, and please enjoy the new gem :)

**Pull request merged**